diff -Naur i2c/i2c_master_bit_ctrl.v i2c_new/i2c_master_bit_ctrl.v
--- i2c/i2c_master_bit_ctrl.v	2013-11-27 18:07:56.246803111 +0000
+++ i2c_new/i2c_master_bit_ctrl.v	2013-11-27 16:37:28.123043869 +0000
@@ -568,7 +568,7 @@
                         c_state <= wr_d;
                         scl_oen_master <= 1'b1; // keep SCL high
                         sda_oen_master <= din;
-                        sda_chk <= 1'b1; // check SDA output
+                        sda_chk <= 1'b0; // check SDA output
                     end
 
                     wr_d:
@@ -726,5 +726,7 @@
     // assign scl and sda output (always gnd)
     assign scl_o = 1'b0;
     assign sda_o = 1'b0;
+    //assign scl_o = 1'b1;
+    //assign sda_o = 1'b1;
 
 endmodule
diff -Naur i2c/i2c_master_top.v i2c_new/i2c_master_top.v
--- i2c/i2c_master_top.v	2013-11-27 18:07:56.258802961 +0000
+++ i2c_new/i2c_master_top.v	2013-11-27 17:54:47.272721859 +0000
@@ -163,20 +163,25 @@
     wb_ack_o <=  wb_cyc_i & wb_stb_i & ~wb_ack_o;
 
 	// assign DAT_O
-	always @(posedge wb_clk_i)
+/*	always @(posedge wb_clk_i)
 	begin
 	  case (wb_adr_i) // synopsys parallel_case
-	    3'b000: wb_dat_o <= prer[ 7:0];
-	    3'b001: wb_dat_o <= prer[15:8];
-	    3'b010: wb_dat_o <= ctr;
-	    3'b011: wb_dat_o <= rxr; // write is transmit register (txr)
-	    3'b100: wb_dat_o <= sr;  // write is command register (cr)
-	    3'b101: wb_dat_o <= txr; // Debug out of TXR
-	    3'b110: wb_dat_o <= cr;  // Debug out control reg
+	    //3'b000: wb_dat_o <= prer[ 7:0];
+	    //3'b001: wb_dat_o <= prer[15:8];
+	   // 3'b010: wb_dat_o <= ctr;
+	  //  3'b011: wb_dat_o <= rxr; // write is transmit register (txr)
+	    //3'b100:  // recive
+			begin
+			 	wb_dat_o <= rxr;
+			 	cr <= 8'b0010000;
+			end
+
+	    //3'b101: wb_dat_o <= txr; // Debug out of TXR
+	    //3'b110: wb_dat_o <= cr;  // Debug out control reg
 	    3'b111: wb_dat_o <= {1'b0,sladr};   // slave address register
 	  endcase
 	end
-
+*/
 	// generate registers
 	always @(posedge wb_clk_i or negedge rst_i)
 	  if (!rst_i)
@@ -196,11 +201,23 @@
 	  else
 	    if (wb_wacc)
 	      case (wb_adr_i) // synopsys parallel_case
-	         3'b000 : prer [ 7:0] <= wb_dat_i;
-	         3'b001 : prer [15:8] <= wb_dat_i;
-	         3'b010 : ctr         <= wb_dat_i;
-	         3'b011 : txr         <= wb_dat_i;
-	         3'b111 : sladr       <=  wb_dat_i[6:0];
+	         3'b000 :
+			begin
+			 	prer [ 7:0] <= wb_dat_i;
+			 	cr <= 8'b10010000;
+			end
+	     /*    3'b001 :
+			begin
+			 	prer [15:8] <= wb_dat_i;
+			 	cr <= 8'b00010000;
+			end */
+	       //  3'b010 : ctr         <= wb_dat_i;
+	         3'b011 : //send
+			begin
+			 	txr <= wb_dat_i;
+			 	cr <= 8'b00010000;
+			end
+	        // 3'b111 : sladr       <=  wb_dat_i[6:0]; //send slave addr
 	         default: ;
 	      endcase
 
